src/components/DeckMechanisms/index.js

```diff
@@ -367,28 +381,36 @@ export default class DeckMechanisms extends React.Component {
       // Deal with active Frozen Cores depending on whether RNG is friendly,
       // unfriendly or regular
       if (state.RNG === 'UNFRIENDLY') {
-        newState.specifics.activeFrozenCores -= 1
+        newState.specifics.activeFrozenCores = 0
       } else if (state.RNG === 'REGULAR') {
         const { activeFrozenCores } = newState.specifics
 
         newState.specifics.activeFrozenCores = Array.from(
           { length: activeFrozenCores },
-          _ => Math.random() >= 0.5
+          _ => Math.random() <= FROZEN_CORE_STAYS
         ).filter(Boolean).length
       }
 
       // If there are some active Frozen Cores, increment the new available mana
       newState.mana += newState.specifics.activeFrozenCores * 3
 
-      if (newState.specifics.liveDawnsparks) {
-        newState.mana +=
-          state.RNG === 'FRIENDLY' ||
-          (state.RNG === 'REGULAR' && Math.random() >= 0.5)
-            ? 4
-            : 0
-        newState.specifics.liveDawnsparks = false
+      if (state.RNG === 'UNFRIENDLY') {
```

Comment:
We could group all these checks:

```js
switch (state.RNG) {
  case 'UNFRIENDLY': {
    // Deal with Dawnsparks
    // Deal with Frozen Core
    // Deal Ahmi
    // …
  }
  
  case 'REGULAR': {
    // …
  }
}
```

Reply:
We could even move that in a separate method. Something like `resolveRNG`? I don’t know if that would make sense. What do you think?